src/Client/ClientBase.cpp

```diff
@@ -868,11 +876,68 @@ void ClientBase::processTextAsSingleQuery(const String & full_query)
         processError(full_query);
 }
 
+String ClientBase::getTableEngine(const String & database, const String & table)
```

Comment:
If we put the logic on the server-side, it will simplify a lot.

Reply:
> it will simplify a lot

Yes, but it will require intoruducing a new server setting for it, like `ignore_drop_queries` and check it in `InterpreterDropQuery`. Will it be ok?

Reply:
Yes, it will be totally ok. We already have fault injection settings, this will be in-line with them.

Reply:
I even planned to move the AST Fuzzer to server side...

Reply:
Done